src/main/webapp/app/exercises/programming/manage/services/programming-exercise.service.ts

```diff
@@ -56,6 +56,14 @@ export class ProgrammingExerciseService {
         return this.http.put<string>(this.resourceUrl + '/' + exerciseId + '/generate-tests', { responseType: 'text' });
     }
 
+    /**
+     * Recreates the BASE and SOLUTION build plan for this exercise
+     * @param exerciseId of the programming exercise for which the build plans should be recreated
+     */
+    recreateBuildPlans(exerciseId: number): Observable<string> {
+        return this.http.put<string>(this.resourceUrl + '/' + exerciseId + '/recreate-build-plans', { responseType: 'text' });
```

Comment:
Could use the inline values of the string interpolation in the literal.

Reply:
Fixed in https://github.com/ls1intum/Artemis/pull/3457/commits/db354dd219003f0c2202f3a4478708e37b7bf533 together with some other requests in this file